include/eve/module/algo/algo/transform_copy_if.hpp

```diff
@@ -0,0 +1,164 @@
+        }
+
+        EVE_FORCEINLINE
+        bool step_1(auto f)
+        {
+          load_apply_func_and_store(f, eve::ignore_none, eve::ignore_none);
+          return false;
+        }
+
+        EVE_FORCEINLINE
+        bool step_2(auto f)
+        {
+          // ol - of < cardinal
+          auto out_ignore   = eve::keep_first(ol - of);
+          load_apply_func_and_store(f, eve::ignore_none, out_ignore);
+          return of == ol;
+        }
+      };
+
+      template<relaxed_range In, relaxed_range Out, typename Func>
+      EVE_FORCEINLINE
+      auto operator()(In&& in, Out&& out, Func func) const -> unaligned_iterator_t<Out>
+      {
+        if( in.begin() == in.end() || out.begin() == out.end() )
+          return out.begin();
+
+        auto [processed_in, processed_out] = temporary_preprocess_ranges_hack(
+          TraitsSupport::get_traits(), in, out);
+
+        auto iteration = two_stage_iteration(processed_in.traits(), processed_in.begin(), processed_in.end());
+
+        auto of = unalign(processed_out.begin());
+        auto ol = unalign(processed_out.end());
+
+        delegate<decltype(of), Func> d {of, ol, func};
+        iteration(d);
+        return eve::unalign(out.begin()) + (d.of - processed_out.begin());
+      }
+
+      template<relaxed_range In, relaxed_range Out, typename Op, typename P>
+      EVE_FORCEINLINE
+      auto operator()(In&& in, Out&& out, Op op, P p) const -> unaligned_iterator_t<Out>
+      {
+        return operator()(in, out, [&](auto elt) {
+          return kumi::make_tuple(op(elt), p(elt));
+        });
+      }
+    };
+  }
+
+  //==============================================================================================
+  //! @addtogroup algos
+  //! @{
+  //!   @var transform_copy_if
+  //!
+  //!   **Defined in header**
+  //!
+  //!   @code
+  //!   #include <eve/module/algo.hpp>
+  //!   @endcode
+  //!
+  //!   @brief Similar to applying `eve::transform_to` and `eve::copy_if` at the same time.
+  //!
+  //!   Conditionally copies values from an input range to an output range,
+  //!   transforming them in the process.
+  //!
+  //!   If the output range is too small, fills all the available space and then stops.
+  //!   
+  //!   If the output range's element type is different from the type of the values returned by the transforming function, performs the appropriate conversions.
+  //!
+  //!   There are two overloads: one that takes in two separate operation and predicate functions,
+  //!   and one that takes in a single function that returns a pair.
+  //!   * The first form gives the same input to both functions.
+  //!     You can think of it as applying `eve::copy_if` and then `eve::transform_to`.
+  //!   * The second form is especially useful when the predicate and the transforming operation
+  //!     have some overlap or depend on each other.
+  //!
+  //!   @groupheader{Callable Signatures}
+  //!
+  //!   @code
+  //!   namespace eve::algo
+  //!   {
```

Comment:
Misses name of the function

Reply:
when you say "try this one" do you mean passing a converting view to preprocess_ranges or to two_stage_iteration? i can try this later tonight

Reply:
I meant to figure out what to convert to once. I think it'll look nicer

Reply:
You can't pass converting virw to preprocessing range, you don't know the type